### PR TITLE
Prove sms posession button reordering

### PR DESCRIPTION
**Pull request by relaxnow**
https://www.pivotaltracker.com/n/projects/1163646/stories/100299518 task 8 and 9

### CODE REVIEW

**Comment by relaxnow** [comment]
screen shot 2016-06-14 at 13 56 44

File: src/Surfnet/StepupRa/RaBundle/Resources/views/Vetting/Sms/provePossession.html.twig [hunk 1]
```diff
@@ -20,10 +20,35 @@
...
+    <div class="row">&nbsp;</div>
+    <div class="row offset-top">
+        <div class="col-md-1">
+            &nbsp;
+        </div>
+        <div class="col-md-5">
+            {{ form_row(form.resendChallenge) }}
+        </div>
+        <div class="col-md-3">
+            <button onclick="redirectWithPostTo({{ path('ra_vetting_cancel', { procedureId: app.request.attributes.get('procedureId') })|json_encode }}); return false" class="btn btn-danger pull-right" formnovalidate="formnovalidate">
```
Comment (rjkip) [inline src/Surfnet/StepupRa/RaBundle/Resources/views/Vetting/Sms/provePossession.html.twig]: As far as I know, until now, JavaScript has only provided progressive enhancement.

Comment (relaxnow) [inline src/Surfnet/StepupRa/RaBundle/Resources/views/Vetting/Sms/provePossession.html.twig]: Fair point, I thought of a better alternative that I just implemented.
You should like this better. Could you rereview?

File: src/Surfnet/StepupRa/RaBundle/Controller/Vetting/SmsController.php [hunk 2]
```diff
@@ -112,6 +113,12 @@ public function provePossessionAction(Request $request, $procedureId)
             ->createForm('ra_verify_phone_number', $command, ['procedureId' => $procedureId])
             ->handleRequest($request);
 
+        /** @var SubmitButton $cancelButton */
+        $cancelButton = $form->get('cancel');
+        if ($cancelButton->isClicked()) {
+            return $this->forward('SurfnetStepupRaRaBundle:Vetting:cancelProcedure', ['procedureId' => $procedureId]);
```
Comment (rjkip) [inline src/Surfnet/StepupRa/RaBundle/Controller/Vetting/SmsController.php]: Hmm, this `forward` depends on the fact that both actions require a POST, and that the cancelProcedureAction has a procedureId parameter. Yet, it's not very clear, from the point of view of the cancelProcedure action, that it is depended upon in that way. In other words: we would probably encounter errors during run-time should we modify the cancelProcedure in the future.

What about cancelling the procedure from this action? At least the application won't break run-time.

```php
if ($cancelButton->isClicked()) {
    $this->getVettingService()->cancelProcedure($procedureId);
    $this->addFlash('info', $this->get('translator')->trans('ra.vetting.flash.cancelled'));

    return $this->redirectToRoute('ra_vetting_search');
}
```

**Comment by rjkip** [comment]
Good solution, including the button in the form. Hadn't thought of that. If you agree, apply the suggested change. :+1: both ways!

File: src/Surfnet/StepupRa/RaBundle/Resources/views/Vetting/Sms/provePossession.html.twig [hunk 3]
```diff
@@ -20,10 +20,36 @@
...
+    <div class="row">&nbsp;</div>
+    <div class="row offset-top">
+        <div class="col-md-1">
+            &nbsp;
+        </div>
+        <div class="col-md-5">
+            {{ form_row(form.resendChallenge) }}
+        </div>
+        <div class="col-md-3">
+            <button name="ra_verify_phone_number[cancel]" class="btn btn-danger pull-right" formnovalidate="formnovalidate">
```
Comment (rjkip) [inline src/Surfnet/StepupRa/RaBundle/Resources/views/Vetting/Sms/provePossession.html.twig]: Oh, and is it possible to get the `button[name]` from the form element so this is less fragile?

**Pull request merged**